src/components/Picklist/index.js

```diff
@@ -0,0 +1,401 @@
+        } else {
+            nextActiveIndex = (activeOptionIndex - 1) % this.activeChildren.length;
+        }
+        this.setState({
+            activeOptionIndex: nextActiveIndex,
+            activeOptionName: this.activeChildren[nextActiveIndex].name,
+        });
+    }
+
+    handleKeyDownPressed() {
+        const { activeOptionIndex } = this.state;
+        const nextActiveIndex = (activeOptionIndex + 1) % this.activeChildren.length;
+        this.setState({
+            activeOptionIndex: nextActiveIndex,
+            activeOptionName: this.activeChildren[nextActiveIndex].name,
+        });
+    }
+
+    handleKeyEnterPressed() {
+        const { onChange } = this.props;
+        const { activeOptionIndex } = this.state;
+        const { label, name, icon, value } = this.activeChildren[activeOptionIndex];
+        this.closeMenu();
+        return onChange({
+            label,
+            name,
+            icon,
+            value,
+        });
+    }
+
+    handleKeyPressed(event) {
+        const { isOpen } = this.state;
+        if (isOpen) {
+            if (event.keyCode === UP_KEY || event.keyCode === DOWN_KEY) event.preventDefault();
+            if (this.keyHandlerMap[event.keyCode]) {
+                return this.keyHandlerMap[event.keyCode]();
+            }
+        }
+        return null;
+    }
+
+    registerChild(childRef, childProps) {
+        const { childrenRefs } = this.state;
+        const [...nodes] = getChildMenuItemNodes(this.containerRef.current);
+        const newChildrenRefs = insertChildOrderly(childrenRefs, childRef, nodes);
+        const refIndex = newChildrenRefs.indexOf(childRef);
+        const extendedChildProps = {
+            ...childProps,
+            ref: childRef,
+        };
+        const { value } = this.props;
+        if (!isEmpty(value)) {
+            if (isEmpty(this.currentOption) && childProps.name === value.name) {
+                this.currentOption = {
+                    index: refIndex,
+                    optionProps: extendedChildProps,
+                };
+            } else if (!isEmpty(this.currentOption)) {
+                const { index, optionProps } = this.currentOption;
+                this.activeChildren.splice(index, 0, optionProps);
+                this.activeChildren.splice(refIndex, 0, extendedChildProps);
+                const currentOptionIndex = this.activeChildren.findIndex(
+                    child => child.name === value.name,
+                );
+                this.currentOption = {
+                    index: currentOptionIndex,
+                    optionProps: this.activeChildren[currentOptionIndex],
+                };
+                this.activeChildren.splice(currentOptionIndex, 1);
+            } else {
+                this.activeChildren.splice(refIndex, 0, extendedChildProps);
+            }
+        } else {
+            this.activeChildren.splice(refIndex, 0, extendedChildProps);
```

Comment:
here you are mutating this.activeChildren, don't use splice, remember we need to avoid mutations,
for example this code achieve the same without mutation:
```
this.activeChildren = this.activeChildren.concat(extendedChildProps);
```